### PR TITLE
🌱markdown: pin mdl to 0.12 and fix complaints

### PR DESCRIPTION
**Pull request by tuminoid**
**What this PR does / why we need it**:

mdl 0.12 changes unordered list indentation rule default from 2 spaces to 3 spaces due Kramdown requirements.
https://github.com/markdownlint/markdownlint/pull/373

Pin markdownlint image to 0.12 (with SHA), and fix all markdown comply with mdl 0.12.

All other PR at the moment will fail to markdown check failure.

### CODE REVIEW

File: hack/markdownlint.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ else
     --volume "${PWD}:/workdir:ro,z" \
     --entrypoint sh \
     --workdir /workdir \
-    registry.hub.docker.com/pipelinecomponents/markdownlint:latest \
+    registry.hub.docker.com/pipelinecomponents/markdownlint:0.12.0@sha256:0b8f9fcf0410257b2f3548f67ffe25934cfc9877a618b9f85afcf345a25804a2 \
```
Comment (kashifest) [inline hack/markdownlint.sh:22]: why do we need sha here? release version isnt enough?

Comment (kashifest) [reply hack/markdownlint.sh:22]: ok, +1, when we are configuring renovate for this, I hope renovate would understand this ?

**Review by kashifest — APPROVED** [review]

**Review by lentzi90 — APPROVED** [review]

**Pull request merged**